### ISSUE TITLE
Product field does not send properly in email since 3.1 update

### ISSUE THREAD

**Issue by Quay3** [wpninjas/ninja-forms#2710]
***After further testing, this issue only presents when the form is being previewed, but not after it has been published/appended to a page***

After updating to 3.1.1, it appears that the values of the Product field are not cooperating via merge tag. Reproductions steps to demonstrate:

1. Add Product field and submit button to form
2. Set up email notification with {fields_table} or {all_fields_table} merge tag
3. Submit form; email does not contain Product value

-or-

1. Add product field and submit button to form
2. Set up email notification with merge tag for Product field
3. Submit form; email contains merge tag but not Product field value

**Comment by Quay3** [comment]
User report of this error here: https://wpninjas.kustomerapp.com/app/customers/5909dcab5590bd1000b6a9c5/event/5909dcac5590bd1000b6a9c8

**Comment by wmerckx** [comment]
hi, 
I have this issue also on a published page!
value always shows 0.00 in email notifications instead of the selected value
total value is correct however

i've tried duplicating the form and appending the new form to my page but same issue

**Comment by marceltamminga** [comment]
hi,
this is still a very urgent issue. The value is always 0.00 when submitted into an email, also on published pages.
Tested on a clean installation of Wordpress with only Ninja forms 3.1.1 plugin active. This problem renders most of our forms unusable because we use a lot of product fields.

**Comment by danielnr87** [comment]
I'm also experiencing this issue on a published form on a live page. Product fields displays 0.00 in the emails. The total is correct however. Hopefully this will be updated soon as using products is causing customers to email in with queries. 

Just for your information, I am not using inline qty, but a separate qty field.

Thanks

**Comment by Dwarve** [comment]
Unable to replicate this issue on the newest release. I was able to replicate on an older build per the original post, but current build is emailing correctly. If anyone is still having this issue after updating, please give some additional details to your Wordpress setup and form layout so I can try to replicate.

**Comment by wpnzach** [comment]
@Dwarve did you test with both inline quantity and using a separate quantity field?

**Comment by Dwarve** [comment]
@wpnzach Affirmative. Last test I did just to make sure I wasn't crazy was an unpublished form (on preview page) with 1 inline and 1 separate qty. The email showed both product prices and the total was correct.

**Comment by MindaugasR** [comment]
Hi,

Have same issue. I thought maybe I doing something wrong, but found this issue. I added ~20 products in form, and after submit form receiving all products with 0.00 quantity.

I tried to set price (in fact I don't want to set it, but just tried with 1$), but still same issue. I receiving email with all products same quantity 0.00.

I'm using inline quantity.

Btw, it's in publish page, in preview I get tons of this errors:

Warning: number_format() expects parameter 1 to be float, string given in wp-content/plugins/ninja-forms/includes/Display/Render.php on line 478

Some info from plugin, maybe will help:

Ninja Forms Version: | 3.2.6
WP Version: | 4.9.1 - Supported
WP Multisite Enabled: | No
Web Server Info: | Apache
PHP Version: | 7.1.5
MySQL Version: | 5.6.29
PHP Locale: | negative_sign:
WP Memory Limit: | 40M
WP Debug Mode: | No
WP Language: | Default
WP Max Upload Size: | 32 MB
PHP Post Max Size: | 32M
Max Input Nesting Level: | 64
PHP Time Limit: | 30
PHP Max Input Vars: | 1000
SUHOSIN Installed: | No

**Comment by jbostoen** [comment]
Also experiencing this issue.

Also, in the backend, we do see the correct values. However, for some items, the calculation does not add up. For 2 of the 3 products, the multiplication says 0, while it should be another value.

**Comment by jandebonnet** [comment]
I can confirm the issue here. 
Updated Ninja Forms to latest 3.2.9
2 product fields with inline quantity in my form and 1 calculated total.
I have 2 confirmation emails:

- admin notification with {field:all_fields}
- user confirmation with seperate fields

Both emails show correct calculated total amount, but the separate product amounts are both 0, in both emails.

**Comment by jbostoen** [comment]
I actually got an official reply from the support team. And as a paying customer, I was quite disappointed with it, because I've been a huge fan of Ninja Forms so far.

I don't get why this is one of the very few PHP projects struggling with this, and especially since the **total sum** is correctly calculated; this should not be such a pain to fix.

```
Unfortunately, I was right. This appears to be an instance in which our calculations are failing due to the site locale.

The problem is that the code expects numbers it calculates to be in a standard code format, which uses commas as thousand separators and periods as decimal separators. Our developers attempted to apply a patch some time ago to resolve this issue, but it was ineffective at best, and only served to confuse the issue further.

In order to properly support math in locales that use a comma as a decimal separator, they'll need to rewrite a large portion of our calculations library. It's an issue that our developers are very aware of, but a fix for it isn't anywhere on the immediate horizon.

I'm afraid that this is a shortcoming of our product, and while you could potentially get it to work properly with enough troubleshooting, it's probably best that you look for a different form solution. Ours just isn't built to handle this use case right now.

Let us know how you'd like to proceed.
```

**Comment by mconce** [comment]
I can report same issue.
All product items are reported as 0,00 but total amount is correct.

**Comment by jmcelhaney** [comment]
https://github.com/wpninjas/ninja-forms/issues/3128

**Comment by jmcelhaney** [comment]
https://wpninjas.kustomerapp.com/app/customers/5ab23613b38ea73ff4c5cd17/event/5ab23613bd5715530968dc04

**Comment by EmbDclic** [comment]
Same issue here. I even tried to set my currency to US Dollars to bypass the bug, but it's the same. All products are at 0.00, but the total is correct (my WordPress is in French).

Does anyone have a workaround while waiting for the team to solve the problem?

Thank you!

**Comment by jmcelhaney** [comment]
https://wpninjas.kustomerapp.com/app/customers/5b0407727aa7a16623dce154/event/5b0407727aa7a16eabdce158

**Comment by danielnr87** [comment]
Just went to build another form using the Quantity field and surprised to see this is still an issue! Surely being just a number input, it should be able just to output that number on the email?
Re-built an old form from last year and the problem persists.

Is there an update on this? Thanks

**Comment by muiqo** [comment]
So it has been over a year since reporting and we still can't use product-fields since the quantity will show up as zero when using the merge-reporting-tag?

**Comment by EmbDclic** [comment]
Yes, I did a workaround. It's been several months now, so I don't remember perfectly what I did, but as I understand, I put the products in dollars, and added this CSS code : 
```
.total-container span:after,
.nf-product-price:after {
	content: " €";
}
```
Feel free to ask me if you need more information.

**Comment by sebastiano-riva** [comment]
Thanks your workaround is good.
In my case I can't use that because the client change the products many times. And I have to use the {all_fields_table} the avoid the costant change of the "calculated values"

In addition the I don't know why the system add 2 decimals, for example if the product cost 1$ in the mail I have 100,00$.

Some details of my environment:
PHP: 5.5 version
Ninja form: 3.3.20 version
WordPress: 5.0 version

**Comment by simongehrig** [comment]
Bump.

Still the same here.

Ninja Forms Version 3.4.22
WP 5.3.2
